scripts/src/javatests/org/oppia/android/scripts/xml/XmlSyntaxErrorHandlerTest.kt

```diff
@@ -0,0 +1,150 @@
+import org.junit.rules.TemporaryFolder
+import org.xml.sax.SAXParseException
+import java.io.File
+import javax.xml.parsers.DocumentBuilder
+import javax.xml.parsers.DocumentBuilderFactory
+
+/** Tests for [XmlSyntaxErrorHandler]. */
+class XmlSyntaxErrorHandlerTest {
+  private val builderFactory = DocumentBuilderFactory.newInstance()
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("testfiles")
+  }
+
+  @Test
+  fun testXmlErrorHandler_validXml_noErrorShouldBeCollected() {
+    val validXml =
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <shape xmlns:android="http://schemas.android.com/apk/res/android"
+        android:shape="rectangle">
+        <solid android:color="#3333334D" />
+        <size android:height="1dp" />
+      </shape>
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(validXml)
+    val docBuilder = builderFactory.newDocumentBuilder()
+    val xmlSyntaxErrorHandler = XmlSyntaxErrorHandler()
+    docBuilder.setErrorHandler(xmlSyntaxErrorHandler)
+
+    parseXml(docBuilder = docBuilder, file = tempFile)
+
+    val errorList = xmlSyntaxErrorHandler.retrieveErrorList()
+    assertThat(errorList).isEmpty()
+  }
+
+  @Test
+  fun testXmlErrorHandler_invalidXml_errorShouldBeCollected() {
+    val invalidXml =
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <shape xmlns:android="http://schemas.android.com/apk/res/android"
+        android:shape="rectangle">
+        <solid android:color="#3333334D" />
+        <size android:height="1dp" />
+      </shape>>
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(invalidXml)
+    val docBuilder = builderFactory.newDocumentBuilder()
+    val xmlSyntaxErrorHandler = XmlSyntaxErrorHandler()
+    docBuilder.setErrorHandler(xmlSyntaxErrorHandler)
+
+    parseXml(docBuilder = docBuilder, file = tempFile)
+
+    val errorList = xmlSyntaxErrorHandler.retrieveErrorList()
+    assertThat(errorList).hasSize(1)
+    assertThat(errorList.first().message).isEqualTo(
+      "Content is not allowed in trailing section."
+    )
+    assertThat(errorList.first().getLineNumber()).isEqualTo(6)
+    assertThat(errorList.first().getColumnNumber()).isEqualTo(9)
+  }
+
+  @Test
+  fun testXmlErrorHandler_invokeWarningCase_errorShouldBeCollected() {
+    val xmlSyntaxErrorHandler = XmlSyntaxErrorHandler()
+    xmlSyntaxErrorHandler.warning(SAXParseException("test_error_message", "", "", 1, 1))
+    val errorList = xmlSyntaxErrorHandler.retrieveErrorList()
+    assertThat(errorList).hasSize(1)
+    assertThat(errorList.first().message).isEqualTo("test_error_message")
+    assertThat(errorList.first().getLineNumber()).isEqualTo(1)
+    assertThat(errorList.first().getColumnNumber()).isEqualTo(1)
+  }
+
+  @Test
+  fun testXmlErrorHandler_invokeErrorCase_errorShouldBeCollected() {
```

Comment:
It feels a bit like fatal error is missing here. Maybe add a separate test for it even though it's technically covered in parsing test above (consider that implementation changes may result in the above error no longer being a fatal error which technically would reduce behavioral coverage in this test suite).

Reply:
Added.